examples/platform/silabs/zephyr/source/AppTaskBase.cpp

```diff
@@ -0,0 +1,478 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    Copyright (c) 2021 Google LLC.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "AppTaskBase.h"
+#include "AppEvent.h"
+#include "CHIPDeviceManager.h"
+#include "CommonDeviceCallbacks.h"
+
+#include <app/server/Dnssd.h>
+#include <lib/dnssd/Advertiser.h>
+
+#include <app/util/attribute-storage.h>
+#include <data-model-providers/codegen/Instance.h>
+#include <setup_payload/OnboardingCodesUtil.h>
+
+#include <app/clusters/network-commissioning/network-commissioning.h>
+
+#include <platform/CommissionableDataProvider.h>
+
+#include "lib/core/ErrorStr.h"
+
+#include <platform/DeviceInstanceInfoProvider.h>
+
+#include <DeviceInfoProviderImpl.h>
+#include <credentials/DeviceAttestationCredsProvider.h>
+#include <credentials/examples/DeviceAttestationCredsExample.h>
+#include <platform/CommissionableDataProvider.h>
+#include <platform/DeviceInstanceInfoProvider.h>
+#include <platform/Zephyr/DeviceInstanceInfoProviderImpl.h>
+
+/* Flag generated by Zap */
```

Comment:
This means you will need zap includes that are app specific in this source. That seems not ideal. Can we have some other way to configure this?

I imagine this is why you include <attribute-storage.h> however this also means the entire app is coupled to zap. Could we separate out the app-specific files more strongly?

Reply:
done, moved to the application part